### PR TITLE
[TwoFluids] Processes Call Order

### PR DESCRIPTION
**Pull request by mrhashemi**
**Description**
Addressing Issue #8346.

Please mark the PR with appropriate tags: 
- Python Solver

**Changelog**
Please summarize the changes in one list to generate the changelog:
- Reordering the calling of the processes in two-fluids Navier-Stokes solver.

### CODE REVIEW

**Comment by mrhashemi** [comment]
@rubenzorrilla once we conclude the discussion, we can implement/merge it.

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
I'd just modify the order of what we have so far. Introducing the distance and mass into the solver deserves a PR on its own.

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py [hunk 1]
```diff
@@ -73,7 +73,23 @@ def GetDefaultParameters(cls):
...
+                "log_file_name"                          : "mass_conservation.log"
+            },
+            "distance_modification_settings": {
+                "apply_distance_modification"            : true
+                "distance_threshold"                     : 0.01,
+                "continuous_distance"                    : true,
+                "check_at_each_time_step"                : false,
+                "avoid_almost_empty_elements"            : true,
+                "deactivate_full_negative_elements"      : true,
+                "recover_original_distance_at_each_step" : false
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py]:
```suggestion
                "apply_distance_modification"            : true
                "distance_threshold"                     : 1.0e-4,
                "continuous_distance"                    : true,
                "check_at_each_time_step"                : true,
                "avoid_almost_empty_elements"            : false,
                "deactivate_full_negative_elements"      : false
```
These are the suitable defaults for the two-fluids. About the default tolerance, I'd put it smaller.

File: applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_two_fluids_solver.py [hunk 2]
```diff
@@ -64,7 +64,23 @@ def GetDefaultParameters(cls):
...
+            "distance_modification_settings": {
+                "apply_distance_modification"            : true
+                "distance_threshold"                     : 0.01,
+                "continuous_distance"                    : true,
+                "check_at_each_time_step"                : false,
+                "avoid_almost_empty_elements"            : true,
+                "deactivate_full_negative_elements"      : true,
+                "recover_original_distance_at_each_step" : false
+            }
         }""")
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_two_fluids_solver.py]: Check #8364. Finally we will get rid of these :smile:

Comment (mrhashemi) [reply applications/FluidDynamicsApplication/python_scripts/trilinos_navier_stokes_two_fluids_solver.py]: Removed for now.

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py (hunk 1 above)
Comment (mrhashemi) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py]: Removed for this PR.

**Comment by mrhashemi** [comment]
@rubenzorrilla the only change in the navier_stokes_two_fluids_solver.py was corresponding the re-distancing.

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py [hunk 3]
```diff
@@ -245,6 +226,25 @@ def InitializeSolutionStep(self):
     def FinalizeSolutionStep(self):
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Mass and momentum conservation equations are solved.")
 
+        # Recompute the distance field according to the new level-set position
+        if (self._reinitialization_type == "variational"):
+            self._GetDistanceReinitializationProcess().Execute()
+        elif (self._reinitialization_type == "parallel"):
+            adjusting_parameter = 0.05
+            layers = int(adjusting_parameter*self.main_model_part.NumberOfElements()) # this parameter is essential
```
Comment (rubenzorrilla) [inline applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py:234]: Just wondering, why so many layers? I think that in the past we used to use something around 10.

**Review by rubenzorrilla — APPROVED** [review]
Thanks! Fine if tests are OK.

File: applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py (hunk 3 above)
Comment (mrhashemi) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py:234]: I have normalized the number of layers with the number of elements. Unfortunately, this re-distancing scheme needs too many iterations in order to get a desirable result.

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py:234]: Actually we cannot ensure a priori that the resultant distance is real distance "enough" for the next convection right?

Comment (mrhashemi) [reply applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py:234]: Yes. That is the point. As far as I checked, if the level-set function is not (let's say) regular enough over the whole domain, the result would be affected.

**Pull request merged**